cpp/vcpkg.json

```diff
@@ -49,6 +54,7 @@
     "rapidjson",
     "re2",
     "snappy",
+    "spdlog",
```

Comment:
Not necessarily for this PR, but I think we should consider whether we want to bring in another logging framework (though I suppose glog probably can't handle what you'd like to do with spdlog)

Reply:
Yea `flightsql-odbc` had already used `spdlog`, but that's something to consider for sure